sklearn/gaussian_process/kernels.py

```diff
@@ -1861,3 +1861,181 @@ def is_stationary(self):
     def __repr__(self):
         return "{0}(gamma={1}, metric={2})".format(
             self.__class__.__name__, self.gamma, self.metric)
+
+
+class SelectDimensionKernel(Kernel):
```

Comment:
Is "select dimension" certainly the right name? It doesn't select, and we substitute "select" with "active" and"dimension" with "features" below. Is "kernel" necessary? 

I'm pretty sure the literature just uses statements  like "we use the product of a RBF kernel applied to ...".  Is AppyTo or Applied to sufficient?